espresso/core/java/androidx/test/espresso/util/HumanReadables.java

```diff
@@ -72,27 +75,12 @@ public static String getViewHierarchyErrorMessage(
 
     errorMessage.append("\n\nView Hierarchy:\n");
 
-    Joiner.on("\n")
-        .appendTo(
-            errorMessage,
-            Iterables.transform(
-                depthFirstViewTraversalWithDistance(rootView),
-                new Function<ViewAndDistance, String>() {
-                  @Override
-                  public String apply(ViewAndDistance viewAndDistance) {
-                    String formatString = "+%s%s ";
-                    if (problemViews != null && problemViews.contains(viewAndDistance.getView())) {
-                      formatString += problemViewSuffix;
-                    }
-                    formatString += "\n|";
-
-                    return String.format(
-                        Locale.ROOT,
-                        formatString,
-                        Strings.padStart(">", viewAndDistance.getDistanceFromRoot() + 1, '-'),
-                        HumanReadables.describe(viewAndDistance.getView()));
-                  }
-                }));
+    ViewHierarchyRenderer.ProblemViews problemViewsToHighlight = null;
+    if (problemViews != null) {
+      problemViewsToHighlight =
+          new ViewHierarchyRenderer.ProblemViews(problemViewSuffix, problemViews);
+    }
+    errorMessage.append(Espresso.getViewHierarchyRenderer().render(rootView, problemViewsToHighlight));
```

Comment:
There's no direct access to the graph from here (different package) and this is a static method used by static methods. So I figured I'd expose UpdatableViewHierarchyRenderer as an Espresso static method, which let me retrieve it from here and allows consumers to manually update by calling it directly.

An alternate approach would be to move HumanReadable to `androidx.test.espresso` to make it possible to call `GraphHolder.baseLayer()`. I didn't do that because the only caller is Espresso.

To be honest it looks like there's a mix of different patterns for how to do things in here so I'm not sure which one is the most up to date. 

UpdatableViewHierarchyRenderer doesn't strongly need to belong to the dagger graph, it could also be its own self managed singleton 🤷‍♂️

Reply:
Whether it is possible to load the renderer implementation within this HumanReadables class?
Currently, the HumanReadables is more like a helper util class that we use everywhere in Espresso for view rendering. It appears to me that in the long run, we might follow a similar approach as the `FailureHandler` that we can configure different `FailureHandler`s for the `ViewInteraction`, but that requires major refactoring in the codebase to replace the static reference to `HumanReadables`, which might be a bit out of scope of this PR. 
@brettchabot WDYT?